### PR TITLE
/rooms/{roomId}/bots/{botId}/data Route Works with Bot name or id

### PR DESCRIPTION
**Pull request by kriscfoster**
This route now works with the name or the id of the bot.
Also added a test for this.

### CODE REVIEW

File: api/v1/controllers/botData.js [hunk 1]
```diff
@@ -56,7 +57,22 @@ module.exports = {
    * @param {Function} next - The next middleware function in the stack
    */
   findBotData(req, res, next) {
-    doFind(req, res, next, helper);
+    if (req.swagger.params.botId) {
```
Comment (tausifmuzaffar) [inline api/v1/controllers/botData.js]: if(!looksLikeId(botId)) then

File: tests/api/v1/botData/get.js [hunk 2]
```diff
@@ -188,6 +188,36 @@ describe('tests/api/v1/botData/get.js >', () => {
     .catch(done);
   });
 
+  it('Pass, get data by room and bot name', (done) => {
```
Comment (tausifmuzaffar) [inline tests/api/v1/botData/get.js]: Add failing test

Comment (kriscfoster) [reply tests/api/v1/botData/get.js]: Just added a failing test at the bottom of the tests but it didn't clear this comment.

**Comment by tausifmuzaffar** [comment]
@kriscfoster update branch but i approved

**Pull request merged**